tests/attribute_tests.cpp

```diff
@@ -151,4 +151,51 @@ BOOST_AUTO_TEST_CASE(IpAddressClone)
     BOOST_CHECK_EQUAL(cs->type(), 4);
 }
 
+BOOST_AUTO_TEST_CASE(EncryptedDataConstructor)
+{
+    std::array<uint8_t, 16> auth {0x92, 0xfa, 0xa1, 0xed, 0x98, 0x9b, 0xb4, 0x79, 0xfe, 0x20, 0xe2, 0xf4, 0x7f, 0x4a, 0x5a, 0x70};
+    std::vector<uint8_t> d {0x25, 0x38, 0x58, 0x18, 0xae, 0x97, 0xeb, 0xeb, 0xbd, 0x46, 0xfd, 0xb9, 0xd1, 0x17, 0x84, 0xeb};
+    RadProto::Encrypted s(2, d.data(), d.size(), "secret", auth);
+
+    BOOST_CHECK_EQUAL(s.toString(), "123456");
+
+    std::vector<uint8_t> values = s.toVector("secret", auth);
+    std::vector<uint8_t> expected {0x02, 0x12, 0x25, 0x38, 0x58, 0x18, 0xae, 0x97, 0xeb, 0xeb, 0xbd, 0x46, 0xfd, 0xb9, 0xd1, 0x17, 0x84, 0xeb};
```

Comment:
`expected` should contain all `d` elements prefixed with `0x02, <length>`. How about making a copy of `d` in elements and using `push_front`? Or reserving a full length, putting `0x02, <length>` inside, and then using `std::copy`?